aten/src/ATen/core/NamedTensor.cpp

```diff
@@ -13,6 +13,11 @@ bool NamedTensorMeta::has_names() const {
       });
 }
 
+/// thread_local is a feature that is not enabled by Caffe2 mobile
+/// build (e.g. iOS). Therefore, we only provide `at::NamesMode`
+/// when we are not in mobile build or when FEATURE_TORCH_MOBILE
+/// is on.
```

Comment:
is there any alternative? I'd be concerned someone might write non-mobile code with named tensors and then want to use it on mobile later. porting to mobile would then require removing all the names.

Reply:
We have the same dilemma with grad_mode: https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/core/grad_mode.cpp.

I think it's fine for now to say that "named tensors are not supported on mobile", just like how "named tensors are not supported on JIT" yet. 

As for alternatives, previous versions of iOS doesn't support thread-local variables, but newer versions do. Android does support thread_local.